### PR TITLE
Add `userAuthenticationEmail` to 1-Time Checkout

### PR DESCRIPTION
**Pull request by scannillo**
### Background

- See Android PR https://github.com/braintree/braintree_android/pull/978/files

### Summary of changes

- Add `userAuthenticationEmail` property to `BTPayPalCheckoutRequest`
- Expose text field for email in Demo
   - Cherry-picked demo app UI updates from this App Switch PR (https://github.com/braintree/braintree_ios/pull/1273)

### Demo UI

Screenshot 2024-04-18 at 11 13 39 AM

### Checklist

- [X] Added a changelog entry

### Authors
@scannillo

### CODE REVIEW

File: Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift [hunk 1]
```diff
@@ -152,6 +155,7 @@ import BraintreeCore
             checkoutParameters["postal_code"] = shippingAddressOverride?.postalCode
             checkoutParameters["country_code"] = shippingAddressOverride?.countryCodeAlpha2
             checkoutParameters["recipient_name"] = shippingAddressOverride?.recipientName
+            checkoutParameters["recipient_email"] = userAuthenticationEmail
```
Comment (jaxdesmarais) [inline Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift:158]: For sure not in scope from our side but it's odd to me that on the gateway side we call this `payer_email` and for checkout we call it `recipient_email`

Comment (scannillo) [reply Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift:158]: It should be `payer_email`, updated!

**Comment by scannillo** [comment]
Update the base branch to be for the ShopperInsights feature based on convos that E2E testers will be testing this alongside ShopperInsights and the 2 features are "planned" to be released together.

File: Demo/Application/Features/PayPalWebCheckoutViewController.swift [hunk 2]
```diff
@@ -5,33 +5,68 @@ import BraintreePayPal
...
+        
         let lineItem = BTPayPalLineItem(quantity: "1", unitAmount: "5.00", name: "item one 1234567", kind: .debit)
         lineItem.upcCode = "123456789"
         lineItem.upcType = .UPC_A
         lineItem.imageURL = URL(string: "https://www.example.com/example.jpg")
         request.lineItems = [lineItem]
+        
+        if payLaterToggle.isOn {
+            request.offerPayLater = true
+        }
```
Comment (jaxdesmarais) [inline Demo/Application/Features/PayPalWebCheckoutViewController.swift:69]: Take it or leave it:
```suggestion
        request.offerPayLater = payLaterToggle.isOn ? true : false
```

File: Demo/Application/Features/PayPalWebCheckoutViewController.swift [hunk 3]
```diff
@@ -63,24 +101,21 @@ class PayPalWebCheckoutViewController: PaymentButtonBaseViewController {
...
-
-            self.completionBlock(nonce)
-        }
+    
+    // MARK: - Helpers
+    
+    private func buttonsStackView(label: String, views: [UIView]) -> UIStackView {
+        let titleLabel = UILabel()
+        titleLabel.text = label
+        titleLabel.font = UIFont.preferredFont(forTextStyle: .title3)
```
Comment (jaxdesmarais) [inline Demo/Application/Features/PayPalWebCheckoutViewController.swift:110]:
```suggestion
        titleLabel.font = .preferredFont(forTextStyle: .title3)
```

**Review by jaxdesmarais — APPROVED** [review]

**Comment by scannillo** [comment]
> Are we seeing the same success in sand and error in prod as we are on Android?

I'm seeing this work in sandbox, but am not seeing the email prefill in production. I don't see a specific error in production though. @tdchow - what error in prod are you seeing?

**Comment by scannillo** [comment]
Failing unit test unrelated `testSendAnalyticsEvent_whenConfigFetchCompletes_setsUpAnalyticsHTTPToUseBaseURL`

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutRequest.swift [hunk 4]
```diff
@@ -10,11 +10,6 @@ import BraintreePayPal
 
 /// Options for the PayPal Checkout flow.
 @objcMembers public class BTPayPalNativeCheckoutRequest: BTPayPalCheckoutRequest {
-
-    // MARK: - Public Properties
-
-    /// Optional: User email to initiate a quicker authentication flow in cases where the user has a PayPal Account with the same email.
-    public var userAuthenticationEmail: String?
```
Comment (agedd) [inline Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutRequest.swift:17]: just making sure, did we mean to remove the `userAuthenticationEmail` from `BTPayPalNativeCheckoutRequest`

**Review by agedd — APPROVED** [review]
lgtm! 🚀

**Comment by sarahkoop** [comment]
> > Are we seeing the same success in sand and error in prod as we are on Android?
> 
> I'm seeing this work in sandbox, but am not seeing the email prefill in production. I don't see a specific error in production though. @tdchow - what error in prod are you seeing?

Confirmed it's expected to not be passed in prod currently (prod has not been released) - should not block merging this PR, we will hold on the full feature merge/release until this is rolled out in prod

**Comment by tdchow** [comment]
> I'm seeing this work in sandbox, but am not seeing the email prefill in production. I don't see a specific error in production though. @tdchow - what error in prod are you seeing?

Apologies! I wasn't seeing a specific error, but I was not seeing email prefilled in the live environment, same as iOS.

**Pull request merged**